src/QsCompiler/CompilationManager/ProjectManager.cs

```diff
@@ -1431,6 +1431,35 @@ editCollection switch
             return diagnostics;
         }
 
+        /// <summary>
+        /// Returns a boxed bool (or null) representing if <paramref name="textDocument"/>
+        /// is a notebook cell or not.
+        /// </summary>
+        /// <remarks>
+        /// Returns null if the given file is null.
+        /// <para/>
+        /// This method waits for all currently running or queued tasks to finish
+        /// before getting the file content.
+        /// </remarks>
+        public object? FileIsNotebookCell(TextDocumentIdentifier textDocument)
```

Comment:
I think this boxing will also be removed if you change `FileIsNotebookCell` like in my previous comment.